pkg/distributor/distributor.go

```diff
@@ -1125,8 +1125,11 @@ func (d *Distributor) send(ctx context.Context, ingester ring.InstanceDesc, time
 		Source:     source,
 	}
 	_, err = c.Push(ctx, &req)
+	if errors.Is(err, context.DeadlineExceeded) {
+		return errors.Wrap(err, "timed out pushing to ingester")
+	}
```

Comment:
I think "timed out" and "deadline exceeded" are synonyms, so this error message is no improvement.

`PushWithCleanup` is setting up the context where it's clear that timeout can only be caused by configured request timeout value. If you want to express that, I would suggest to 1) move handling of DeadlineExceeded to `PushWithCleanup` (in anonymous function in DoBatch), 2) improve the message to make it clear that it's distributor's configured timeout that caused this.

Reply:
@pstibrany please have a look at the latest revision.